README.md

```diff
@@ -1,5 +1,175 @@
+* `slash authority`: Optional authority that can slash any stake account.
+* `cooldown time seconds`: After a deactivation, defines the number of seconds that must pass before the stake is inactive and able to be withdrawn.
+* `sync rewards lamports`: Lamports amount paid as a reward for syncing a SOL stake account.
+* `maximum deactivation basis points`: The maximum proportion that can be deactivated at once, given as basis points (`1 / 10000`).
+
+Each `Config` account is associated with a particular mint account, determined by the mint of its `vault` token account. The `vault` token account holds all the staked tokens and it is controlled by the `vault authority` of the `Config` account.
+
+> [!NOTE]
+> While staked tokens are escrowed by the `Config` account, they still accrue holder rewards in addition to the staking. There are specific instructions on the program that allows holders to claim both their "holder" and "staking" rewards.
+
+### `SolStakerStake`
+
+The `SolStakerStake` accounts hold the delegation information of individual SOL stakers. The delegation holds the amount of staked tokens as well as their SOL stake state.
+
+The maximum amount of tokens that a SOL staker is allowed to stake is currently proportional to the amount of SOL staked, given by `1.3 * SOL amount staked`.
+
+
+### `ValidatorStake`
+
+The `ValidatorStake` accounts hold the delegation information for the tokens staked by a validator. It also tracks the total amount of SOL and tokens staked by its stakers.
+
+The total amount of SOL staked on a validator is used to determine that maximum amount of tokens that the validator is allowed to stake &mdash; currently the limit is given by `1.3 * SOL amount staked`.
+
+## 📋 Instructions
+
+- [`DeactivateStake`](#deactivatestake)
+- [`DistributeRewards`](#distributerewards)
+- [`InactivateSolStakerStake`](#inactivatesolstakerstake)
+- [`InactivateValidatorStake`](#inactivatevalidatorstake)
+- [`InitializeConfig`](#initializeconfig)
+- [`InitializeSolStakerStake`](#initializesolstakerstake)
+- [`InitializeValidatorStake`](#initializevalidatorstake)
+- [`HarvestHolderRewards`](#harvestholderrewards)
+- [`HarvestSolStakerRewards`](#harvestsolstakerrewards)
+- [`HarvestSyncRewards`](#harvestsyncrewards)
+- [`HarvestValidatorRewards`](#harvestvalidatorrewards)
+- [`SetAuthority`](#setauthority)
+- [`SlashSolStakerStake`](#slashsolstakerstake)
+- [`SlashValidatorStake`](#slashvalidatorstake)
+- [`SolStakerStakeTokens`](#solstakerstaketokens)
+- [`SyncSolStake`](#syncsolstake)
+- [`UpdateConfig`](#updateconfig)
+- [`ValidatorStakeTokens`](#validatorstaketokens)
+- [`WithdrawInactiveStake`](#withdrawinactivestake)
+
+### `DeactivateStake`
+
+Deactivate staked tokens for a stake delegation, either `ValidatorStake` or `SolStakerStake`. Only one deactivation may be in-flight at once, so if this is called with an active deactivation, it will succeed, but reset the amount and timestamp.
```

Comment:
Is it worth adding that deactivating 0 tokens will cancel all deactivations?

Reply:
Good call, definitely worth mentioning.